### ISSUE TITLE
Cannot perform consecutive rename operations on categorized tag

### ISSUE THREAD

**Issue by i-plasm** [freeplane/freeplane#2041]
**Describe the bug**
Cannot perform consecutive rename operations on categorized tag

**To Reproduce**
1. Create a new mind map
1. Add this tag to the root node (by right clicking the node and choosing `Edit -> Edit tags`):
   - `aaa::bbb`
1. Click OK
1. Open the categories manager (`Edit -> Node properties -> Manage tag categories for map`)
1. Rename `aaa` to `kkk`
1. Rename `bbb` to `mmm`
1. =>>> ERROR: The last rename fails and Exception is thrown (see log at the end)
1. Extra info: the error does not happen if the "child" tag `bbb` is renamed first, then the parent `aaa`, so it is tied to order of events

**Expected behavior**
It should be possible to perform consecutive rename operations on any categorized tag subunits, in any order

**Desktop (please complete the following information):**
- freeplane-1.12.6-pre06

**Logs**

~~~
SEVERE: Exception in thread "AWT-EventQueue-0"
java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because "references" is null
	at org.freeplane.features.icon.TagCategories.registerTagReference(TagCategories.java:506)
	at org.freeplane.features.icon.TagCategories.registerTagReference(TagCategories.java:495)
	at org.freeplane.features.icon.TagCategories.createTagReference(TagCategories.java:491)
	at org.freeplane.features.icon.TagCategories.createTag(TagCategories.java:487)
	at org.freeplane.features.icon.TagCategories.categorizedTag(TagCategories.java:751)
	at org.freeplane.features.icon.TagCategories$TagCategoryTree.valueForPathChanged(TagCategories.java:72)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.completeEditing(BasicTreeUI.java:2436)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.editingStopped(BasicTreeUI.java:4200)
	at java.desktop/javax.swing.AbstractCellEditor.fireEditingStopped(AbstractCellEditor.java:152)
	at org.freeplane.features.icon.mindmapmode.TagCategoryEditor$TagCellEditor.stopCellEditing(TagCategoryEditor.java:227)
	at org.freeplane.features.icon.mindmapmode.TagCategoryEditor$TagCellEditor$1.actionPerformed(TagCategoryEditor.java:169)
	at java.desktop/javax.swing.JTextField.fireActionPerformed(JTextField.java:525)
	at java.desktop/javax.swing.JTextField.postActionEvent(JTextField.java:740)
	at java.desktop/javax.swing.JTextField$NotifyAction.actionPerformed(JTextField.java:856)
	at java.desktop/javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1810)
	at java.desktop/javax.swing.JComponent.processKeyBinding(JComponent.java:2947)
	at java.desktop/javax.swing.JComponent.processKeyBindings(JComponent.java:2995)
	at java.desktop/javax.swing.JComponent.processKeyEvent(JComponent.java:2909)
	at java.desktop/java.awt.Component.processEvent(Component.java:6403)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
	at java.desktop/java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1952)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:883)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1150)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:1020)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:848)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4882)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread
~~~

**Comment by dpolivaev** [comment]
Preview 1.12.6_07

**Comment by i-plasm** [comment]
Hello Dimitry. The latest bugs have been fixed - many thanks! I also tested against all the previous bugs I've posted, and it still working well. Hope it helps! Best regards.

**Comment by dpolivaev** [comment]
@i-plasm  I have released 1.12.6 as stable. Actually, you have not only made me aware of the bugs but also helped me to see that I should implement at least some regression tests for this complicated issue. So your contribution to the tool quality is very significant and highly appreciated.
